### PR TITLE
add changepoint api

### PR DESCRIPTION
**Pull request by whatyouthink0**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 1]
```diff
@@ -112,6 +112,92 @@
           }
         }
       }
+    },
+    "/timeseries/detect/changepoint": {
+      "post": {
+        "summary": "Detect changepoint for the entire series",
+        "description": "This operation generates a model using an entire series, each point is detected with the same model. With this method, points before and after a certain point are used to determine whether it is an changepoint. The entire detection can give user an overall status of the time series.",
+        "operationId": "Detection_Changepoint",
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: Detection_ChangePoint

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 2]
```diff
@@ -112,6 +112,92 @@
           }
         }
       }
+    },
+    "/timeseries/detect/changepoint": {
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: "/timeseries/changepoint/detect"

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 3]
```diff
@@ -112,6 +112,92 @@
...
+        "description": "This operation generates a model using an entire series, each point is detected with the same model. With this method, points before and after a certain point are used to determine whether it is an changepoint. The entire detection can give user an overall status of the time series.",
+        "operationId": "Detection_Changepoint",
+        "parameters": [
+          {
+            "name": "body",
+            "in": "body",
+            "description": "Time series points is needed. Advanced model parameters can also be set in the request if needed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ChangePointRequest"
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointRequest may need to change this name

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 4]
```diff
@@ -112,6 +112,92 @@
...
+          }
+        },
+        "x-ms-examples": {
+          "Find anomalies for the entire series in batch example": {
+            "$ref": "./examples/ChangePointDetect.json"
+          }
+        }
+      }
+    },
+    "/timeseries/{timeseriesId}/detect/changepoint": {
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: /timeseries/{timeseriesId}/changepoint/detect

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 5]
```diff
@@ -112,6 +112,92 @@
...
+        "description": "detect last continuous points which haven't been detected before between begin and end time stamp, for example, if the last changepoint is time t, between t and end, there are N points, but between begin and end, there are M points, in this API, only N points will be detected.",
+        "operationId": "Detection_ChangepointStateful",
+        "parameters": [
+          {
+            "name": "body",
+            "in": "body",
+            "description": "begin and end is needed. Advanced model parameters can also be set in the request if needed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ChangepointStatefulRequest"
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: change name

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 6]
```diff
@@ -112,6 +112,92 @@
...
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful operation.",
+            "schema": {
+              "$ref": "#/definitions/ChangepointStatefulDetectResponse"
```
Comment (moreOver0) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: change name

**Review by moreOver0 — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 7]
```diff
@@ -112,6 +112,92 @@
           }
         }
       }
+    },
+    "/timeseries/changepoint/detect": {
+      "post": {
+        "summary": "Detect change point for the entire series",
+        "description": "This operation generates a model using an entire series, each point is detected with the same model. With this method, points before and after a certain point are used to determine whether it is an change point. The entire detection can give user an overall status of the time series.",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: is this description can be used for change point detection?

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 8]
```diff
@@ -112,6 +112,92 @@
...
+        "operationId": "Detection_ChangePoint",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePoint_Detect

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 9]
```diff
@@ -112,6 +112,92 @@
...
+        "parameters": [
+          {
+            "name": "body",
+            "in": "body",
+            "description": "Time series points and granularity is needed. Advanced model parameters can also be set in the request if needed.",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json:510]: Description

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 10]
```diff
@@ -112,6 +112,92 @@
...
+          "Find anomalies for the entire series in batch example": {
+            "$ref": "./examples/ChangePointDetect.json"
+          }
+        }
+      }
+    },
+    "/timeseries/{timeseriesId}/changepoint/detect": {
+      "post": {
+        "summary": "Detect change point for the interval series",
+        "description": "detect last continuous points which haven't been detected before between begin and end time stamp, for example, if the last change point is time t, between t and end, there are N points, but between begin and end, there are M points, in this API, only N points will be detected.",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: Description

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 11]
```diff
@@ -112,6 +112,92 @@
...
+        "operationId": "Detection_ChangePointStateful",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: TimeSeries_ChangePoint_Detect_InTimeRange

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 12]
```diff
@@ -112,6 +112,92 @@
...
+        "parameters": [
+          {
+            "name": "body",
+            "in": "body",
+            "description": "begin and end is needed. Advanced model parameters can also be set in the request if needed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ChangePointDetectionRequest"
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointDetectInTimeRangeRequest

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 13]
```diff
@@ -112,6 +112,92 @@
...
+        "description": "This operation generates a model using an entire series, each point is detected with the same model. With this method, points before and after a certain point are used to determine whether it is an change point. The entire detection can give user an overall status of the time series.",
+        "operationId": "Detection_ChangePoint",
+        "parameters": [
+          {
+            "name": "body",
+            "in": "body",
+            "description": "Time series points and granularity is needed. Advanced model parameters can also be set in the request if needed.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/InputSeriesChangePointDetectionRequest"
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointDetectRequest

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 14]
```diff
@@ -112,6 +112,92 @@
...
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "#/definitions/APIError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Find anomalies for the entire series in batch example": {
+            "$ref": "./examples/ChangePointStatefulDetect.json"
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointDetectInTimeRange

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 15]
```diff
@@ -336,6 +422,114 @@
           "description": "Anomaly status in positive direction of the latest point. True means the latest point is an anomaly and its real value is larger than the expected one."
         }
       }
+    },
+    "InputSeriesChangePointDetectionRequest":{
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointDetectRequest

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 16]
```diff
@@ -336,6 +422,114 @@
...
+            "type": "string",
+            "format": "date-time"
+          }
+        },
+        "confidence": {
+          "type": "array",
+          "description": "the change point confidence of each point",
+          "items": {
+            "type": "number",
+            "format": "double"
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: float

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json [hunk 17]
```diff
@@ -336,6 +422,114 @@
...
+          }
+        }
+      }
+    },
+    "ChangePointDetectionRequest": {
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/AnomalyDetector.json]: ChangePointDetectInTimeRangeRequest

**Review by conhua — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/examples/ChangePointDetectInTimeRange.json [hunk 18]
```diff
@@ -0,0 +1,799 @@
+{
+  "parameters": {
+    "Endpoint": "{Endpoint}",
+    "Content-Type": "application/json",
+    "Ocp-Apim-Subscription-Key": "{API key}",
+    "timeseriesId":"{timeseriesId}",
```
Comment (conhua) [inline specification/cognitiveservices/data-plane/AnomalyDetector/preview/v1.0/examples/ChangePointDetectInTimeRange.json]: f9f0114c-3c4c-4d3a-a6c7-f4d6d595f4f7

**Review by conhua — APPROVED** [review]

**Pull request merged**